lib/rewards/server/subscribers/rewards-subscriber.js

```diff
@@ -0,0 +1,135 @@
+const { sc, Logger } = require('@reldens/utils');
+
+class RewardsSubscriber
+{
+    static async onBattleEnded(playerSchema, targetObject, events)
+    {
+        let rewards = sc.get(targetObject, 'rewards', false);
+
+        let winningRewards = RewardsSubscriber.getWinningRewards(rewards, false);
```

Comment:
replace all the RewardsSubscriber.method() by this.method():
getWinningRewards
addItemToPlayerInventory
updateUniqueRewards
etc.

Reply:
done!